smartcontract/states/assetstate.go

```diff
@@ -0,0 +1,127 @@
+package states
+
+import (
+	"DNA/common"
+	"DNA/crypto"
+	"io"
+	"DNA/common/serialization"
+	. "DNA/errors"
+	"bytes"
+	"DNA/core/asset"
+)
+
+type AssetState struct {
+	AssetId common.Uint256
+	AssetType asset.AssetType
+	Name string
+	Amount common.Fixed64
+	Available common.Fixed64
+	Precision byte
+	FeeMode byte
+	Fee common.Fixed64
+	FeeAddress *common.Uint160
+	Owner *crypto.PubKey
+	Admin common.Uint160
+	Issuer common.Uint160
+	Expiration uint32
+	IsFrozen bool
+	*StateBase
+}
+
+func(assetState *AssetState)Serialize(w io.Writer) error {
+	assetState.AssetId.Serialize(w)
```

Comment:
forget to serialize the StateBase struct ?

Reply:
it is not necessary in the version。

Reply:
It will cause incompatibility problem when upgrade。

Reply:
next version add statebase do not effect serialize and deserialize